pkgs/os-specific/linux/nvidia-x11/open.nix

```diff
@@ -24,6 +24,11 @@ stdenv.mkDerivation {
     "SYSSRC=${kernel.dev}/lib/modules/${kernel.modDirVersion}/source"
     "SYSOUT=${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
     "MODLIB=$(out)/lib/modules/${kernel.modDirVersion}"
+  ] ++ lib.optionals (stdenv.buildPlatform != stdenv.hostPlatform) [
```

Comment:
It might be a good idea to set this variable during native builds as well, since the makefile otherwise tries to do runtime detection using `uname`: https://github.com/NVIDIA/open-gpu-kernel-modules/blob/6dd092ddb7c165fb1ec48b937fa6b33daa37f9c1/utils.mk#L138

Reply:
Closed-source driver seems to build just fine without the flag. I didn't test the cross-compiled driver though. But I changed it so now that it is always set